include/deal.II/base/vectorization.h

```diff
@@ -1620,12 +2106,36 @@ class VectorizedArray<double>
    * 16 bytes, as opposed to casting a double address to
    * VectorizedArray<double>*.
    */
-  void store (double *ptr) const
+  DEAL_II_ALWAYS_INLINE void store (double *ptr) const
   {
     _mm_storeu_pd (ptr, data);
   }
 
   /**
+   * Gather operation.
+   */
+  DEAL_II_ALWAYS_INLINE void gather (const double *base_ptr,
+                                     const int    *offsets)
+  {
+    for (unsigned int i=0; i<2; ++i)
+      *(reinterpret_cast<double *>(&data)+i) = base_ptr[offsets[i]];
```

Comment:
Out of curiosity, why do you need the cast here? Because `data` is a vectorized type?